src/routes/ExercisesScreens.tsx

```diff
@@ -47,6 +67,17 @@ const ExercisesScreen = ({ route, navigation }: ExercisesScreenProps): JSX.Eleme
 
   return (
     <Root>
+      <ConfirmationModal
+        visible={isModalVisible}
+        setVisible={setIsModalVisible}
+        text="Ups, this exercise is still unlocked."
+        confirmationButtonText="OK,GOT IT!"
+        cancelButtonText="UNLOCK EXERSCISE"
+        confirmationAction={() => null}>
```

Comment:
I think this is the wrong way round.
The `Ok, got it` button should close the modal.
The `Unlock Exercise` button should start the last exercise and therefore trigger a navigation. This can also be done with you `currentLevel` variable.

Also no need to caps-lock in the wording, as our button-component, makes the content of a button caps-lock automatically.